lib/cancel/CancelToken.js

```diff
@@ -35,6 +35,9 @@ function CancelToken(executor) {
  */
 CancelToken.prototype.throwIfRequested = function throwIfRequested() {
   if (this.reason) {
+    if (typeof this.reason === 'string') {
```

Comment:
Not sure if this is the best way of checking string type or if something like this is better:

```js
if (typeof this.reason === typeof 'string')
```